lib/sql.js

```diff
@@ -692,7 +692,7 @@ SQLConnector.prototype._replace = function(model, where, data, options, cb) {
   var stmt = this.buildReplace(model, where, data, options);
   this.execute(stmt.sql, stmt.params, options, function(err, info) {
     if (err) return cb(err);
-    if (info.affectedRows === 0) {
+    if (info.affectedRows === 0 || info.rowsAffected === 0) {
```

Comment:
We should override `getCountForAffectedRows(model, info)` in `loopback-connector-oracle` instead.

See https://github.com/strongloop/loopback-connector/blob/e1db169b8b1871ce5112560f5543e8d7e72a5f94/lib/sql.js#L713.